blockchain/api.go

```diff
@@ -1352,29 +1369,165 @@ func NewSimpleGatekeeper(address common.Address, opts *chainOpts) (SimpleGatekee
 		client:   client,
 		contract: contract,
 		opts:     opts,
+		address:  address,
 	}, nil
 }
 
 func (api *BasicSimpleGatekeeper) PayIn(ctx context.Context, key *ecdsa.PrivateKey, value *big.Int) error {
 	opts := api.opts.getTxOpts(ctx, key, api.opts.gasLimit)
-	tx, err := api.contract.PayIn(opts, value)
+	tx, err := api.contract.Payin(opts, value)
 	if err != nil {
 		return err
 	}
 
-	if _, err := WaitTxAndExtractLog(ctx, api.client, api.opts.blockConfirmations, api.opts.logParsePeriod, tx, PayInTopic); err != nil {
+	if _, err := WaitTxAndExtractLog(ctx, api.client, api.opts.blockConfirmations, api.opts.logParsePeriod, tx, PayinTopic); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (api *BasicSimpleGatekeeper) Payout(ctx context.Context, key *ecdsa.PrivateKey, to common.Address, value *big.Int, txNumber *big.Int) (*types.Transaction, error) {
+func (api *BasicSimpleGatekeeper) GetKeeper(ctx context.Context, keeper common.Address) (*Keeper, error) {
+	k, err := api.contract.Keepers(getCallOptions(ctx), keeper)
+	if err != nil {
+		return nil, err
+	}
+	return &Keeper{
+		Address:    keeper,
+		DayLimit:   k.DayLimit,
+		LastDay:    k.LastDay,
+		SpentToday: k.SpentToday,
+		Frozen:     k.Frozen,
+	}, nil
+}
+
+func (api *BasicSimpleGatekeeper) Payout(ctx context.Context, key *ecdsa.PrivateKey, to common.Address, value *big.Int, txNumber *big.Int) (PayoutResult, error) {
 	opts := api.opts.getTxOpts(ctx, key, api.opts.gasLimit)
-	return api.contract.Payout(opts, to, value, txNumber)
+
+	tx, err := txRetryWrapper(func() (*types.Transaction, error) {
+		return api.contract.Payout(opts, to, value, txNumber)
+	})
+	if err != nil {
+		return 0, err
+	}
+
+	rec, err := WaitTransactionReceipt(ctx, api.client, api.opts.blockConfirmations, api.opts.logParsePeriod, tx)
+	if err != nil {
+		return 0, err
+	}
+
+	for _, l := range rec.Logs {
+		switch l.Topics[0] {
+		case CommitTopic:
+			return Committed, nil
+		case PayoutTopic:
+			return Payouted, nil
+		default:
+			continue
+		}
+	}
+	return UNKNOWN, fmt.Errorf("transaction executed successfuly, but payout's topic not found")
 }
 
 func (api *BasicSimpleGatekeeper) Kill(ctx context.Context, key *ecdsa.PrivateKey) (*types.Transaction, error) {
```

Comment:
BasicSimple looks like a tautology

Reply:
its out of scope of this pr.
I did not set this naming trend.
I'm just follow this.
https://github.com/sonm-io/core/commit/89f23e1afba7ba53be8a0f326156eb3275a8146c#diff-96242c25cfcb524ef908cc32c2a021efL184